### ISSUE TITLE
aws_iot_topic_rule fails to parse SQS in response

### ISSUE THREAD

**Issue by rraub** [terraform-providers/terraform-provider-aws#7334]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

* Terraform v0.11.11
* provider.aws v1.56.0

### Affected Resource(s)

* resource "aws_iot_topic_rule"

### Terraform Configuration Files
```hcl
resource "aws_iot_topic_rule" "hub_messages" {
  name = "hub_messages_${var.environment_name}"
  sql = "SELECT *, topic(1) as serial_number FROM '+/message'"
  sql_version = "2016-03-23"
  sqs {
    queue_url = "${aws_sqs_queue.my_queue.id}"
    role_arn = "${aws_iam_role.my_role.arn}"
    use_base64 = true
  }
}
```

### Debug Output

https://gist.github.com/rraub/fb1d48916ca5606e6048e924461ba641

### Panic Output

```
Error: Error refreshing state: 1 error(s) occurred:

* module.iot.aws_iot_topic_rule.rule_name: 1 error(s) occurred:

* module.iot.aws_iot_topic_rule.rule_name: aws_iot_topic_rule.rule_name: unexpected EOF
```

### Expected Behavior

An IoT Topic Rule with an SQS action can be updated from Terraform.

### Actual Behavior

Terraform AWS provider fails to parse the AWS response in `iot/GetTopicRule`

### Steps to Reproduce

1. Create a terraform aws_iot_topic_rule resource with an SQS action that has a base64 encoding set to false or null.
2. `terraform apply`
3. Change the name of the rule in the terraform configuration
4. `terraform apply`

### Important Factoids

The SQS is also a terraform managed resource, but I don't think that matters.

### References

* https://docs.aws.amazon.com/cli/latest/reference/iot/get-topic-rule.html
* [Line that throws the error in structure.go](https://github.com/terraform-providers/terraform-provider-aws/blob/6fb92ff9d7c0d1a282c5a5ada97b7da54adc0738/aws/structure.go#L2996)

* #0000

**Comment by bflad** [comment]
It looks like the code is unsafely assuming `UseBase64` is always returned in the API response. 

Relevant portion from logs:

```
2019-01-25T10:36:41.021-0700 [DEBUG] plugin.terraform-provider-aws_v1.56.0_x4: 2019/01/25 10:36:41 [DEBUG] [aws-sdk-go] {"rule":{"actions":[{"cloudwatchAlarm":null,"cloudwatchLogs":null,"cloudwatchMetric":null,"dynamoDB":null,"dynamoDBv2":null,"elasticsearch":null,"firehose":null,"iotAnalytics":null,"iotEvents":null,"kinesis":null,"lambda":null,"republish":null,"s3":null,"salesforce":null,"sns":null,"sqs":{"queueUrl":"https://sqs.us-east-1.amazonaws.com/000000000000/queue_name","roleArn":"arn:aws:iam::000000000000:role/service-role/iot-to-sqs","useBase64":null},"stepFunctions":null}],"awsIotSqlVersion":"2016-03-23","createdAt":null,"description":"test","errorAction":null,"ruleDisabled":false,"ruleName":"rule_name","sql":"SELECT *, topic(1) as serial_number FROM '+/message'"},"ruleArn":"arn:aws:iot:us-east-1:000000000000:rule/rule_name"}
2019-01-25T10:36:41.024-0700 [DEBUG] plugin.terraform-provider-aws_v1.56.0_x4: panic: runtime error: invalid memory address or nil pointer dereference
2019-01-25T10:36:41.024-0700 [DEBUG] plugin.terraform-provider-aws_v1.56.0_x4: [signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x3379dba]
2019-01-25T10:36:41.024-0700 [DEBUG] plugin.terraform-provider-aws_v1.56.0_x4: 
2019-01-25T10:36:41.024-0700 [DEBUG] plugin.terraform-provider-aws_v1.56.0_x4: goroutine 183 [running]:
2019-01-25T10:36:41.024-0700 [DEBUG] plugin.terraform-provider-aws_v1.56.0_x4: github.com/terraform-providers/terraform-provider-aws/aws.flattenIoTRuleSqsActions(0xc00037a000, 0x1, 0x1, 0x37124c0, 0xc0008b0a60, 0x0)
2019-01-25T10:36:41.024-0700 [DEBUG] plugin.terraform-provider-aws_v1.56.0_x4: 	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-aws/aws/structure.go:2996 +0x16a
2019-01-25T10:36:41.024-0700 [DEBUG] plugin.terraform-provider-aws_v1.56.0_x4: github.com/terraform-providers/terraform-provider-aws/aws.resourceAwsIotTopicRuleRead(0xc0003a7730, 0x3c9e660, 0xc0001d4380, 0xc0003a7730, 0x0)
2019-01-25T10:36:41.024-0700 [DEBUG] plugin.terraform-provider-aws_v1.56.0_x4: 	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-aws/aws/resource_aws_iot_topic_rule.go:580 +0x9b7
```

**Comment by rraub** [comment]
@bflad Agreed! The aws docs aren't explicit with whats nullable. I'll update the ticket description to include a false or null value for the base64 encoding.

**Comment by bflad** [comment]
Fix submitted. 😄 #7337

**Comment by bflad** [comment]
The fix for this has been merged and will release with version 1.57.0 of the Terraform AWS provider, likely later today. 👍

**Comment by rraub** [comment]
Thanks @bflad !! 🎉

**Comment by bflad** [comment]
This has been released in [version 1.57.0 of the AWS provider](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md#1570-january-26-2019). Please see the [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions) or reach out if you need any assistance upgrading.
